File: Java/src/World/body/plant/Dandelion.java
package World.body.plant;

import GUI.Color_obj;
import World.Point;
import World.World;
import World.body.Plant;

import java.util.concurrent.ThreadLocalRandom;

public class Dandelion extends Plant {
    public Dandelion(Point point_location, World world) {
        super(point_location, world);
        setName("Dandelion");
        setPower(0);
        setColor(Color_obj.DANDELION);
    }

    @Override
    protected void action() {
        int randomNumber;
        for(int i=3;i>0;i--){
            randomNumber = ThreadLocalRandom.current().nextInt(0, 201);
            if(randomNumber%100<8){
                int free_spaces = getWorld().free_spaces(this);
                if(free_spaces>0){
                    Point new_point = new Point(getPoint_location().getX(), getPoint_location().getY());
                    random_location_empty(this, new_point);
                    new_body(new_point);
                    getWorld().getNews_panel().add_news(getName() + " has grown on (" + new_point.getX() + ", " + new_point.getY()+')');
                }
                else{
                    getWorld().getNews_panel().add_news("Nothing happened");
                }
            }
            else{
                getWorld().getNews_panel().add_news("Nothing happened");
            }
        }
    }

    @Override
    public int get_stronger() {
        return 0;
    }
}

File: Python/World/Body/Plant/dandelion.py
from World.Body.plant import Plant
from World.point import Point
from GUI.colors import Colors
from random import randint


class Dandelion(Plant):
    def __init__(self, point_location, world):
        super().__init__(point_location, world)
        self.set_name("Dandelion")
        self.set_power(0)
        self.set_color(Colors.DANDELION)

    def action(self):
        for i in range(3, 0, -1):
            random_number = randint(0, 200)
            if random_number % 100 < 3:
                free_spaces = self.world.free_spaces(self)
                if free_spaces > 0:
                    new_point = Point(self.get_point_location().get_x(), self.get_point_location().get_y())
                    self.random_location_empty(self, new_point)
                    self.new_body(new_point)
                    self.world.mainWindow.add_text(self.get_name() + " has grown on ({}, {})".format(new_point.get_x(), new_point.get_y()))
                else:
                    self.world.mainWindow.add_text("Nothing happened")
            else:
                self.world.mainWindow.add_text("Nothing happened")

    def get_stronger(self):
        return 0
